src/Todo/Bundle/ListBundle/Controller/Api/ItemController.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace Todo\Bundle\ListBundle\Controller\Api;
+
+use FOS\RestBundle\Controller\FOSRestController;
+use Nelmio\ApiDocBundle\Annotation\ApiDoc;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+use Todo\Bundle\ListBundle\Entity\Item;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use FOS\RestBundle\Controller\Annotations as Rest;
+use Todo\Bundle\ListBundle\Entity\ListItem;
+use Todo\Bundle\ListBundle\Form\ItemType;
+
+class ItemController extends FOSRestController
+{
+    /**
+     * Create new Item in specific List
+     *
+     * @Rest\Post("/api/list/{id}/item")
+     * @Rest\View(serializerGroups={"Details", "Default"})
+     * @ApiDoc(
+     *     section="Item",
+     *     input="Todo\Bundle\ListBundle\Form\ItemType"
+     * )
+     *
+     * @param Request $request
+     * @param ListItem $list
+     * @return mixed
+     */
+    public function addItemAction(Request $request,ListItem $list)
+    {
+        if (!$this->getUser() || $list->getUser() != $this->getUser()) {
+            throw new AccessDeniedException("Access Denied");
+        }
+        $em = $this->getDoctrine()->getManager();
+
+        if (!$list){
+            throw $this->createNotFoundException('List is not found');
+        }
+        $item = new Item();
+
+        $form = $this->createForm(ItemType::class, $item, array(
+            'method' => 'POST',
+            'csrf_protection' => false,
+        ));
+
+        $form->handleRequest($request);
+
+        if ($request->isMethod('POST') && $form->isValid()){
+            $em->persist($item);
```

Comment:
Before persisting the item object, are you sure that you assigned that object to it's list?